misc/version_updater.py

```diff
@@ -467,14 +467,18 @@ def check_tool(tool: Tool, update: bool) -> VersionUpdateResult:
         if regexes is None:
             continue
 
-        for glob_file in glob.glob(str(filename), recursive=True):
-            file = Path(glob_file)
-            if update:
-                res = update_tool_version(file, regexes, version)
-                update_res.errors += res.errors
-                update_res.updated.update(res.updated)
-            else:
-                update_res.errors += check_tool_version(file, regexes, version).errors
+        matched_files = glob.glob(str(filename), recursive=True)
+        if len(matched_files) > 0:
+            for file in matched_files:
+                file = Path(file)
+                if update:
+                    res = update_tool_version(file, regexes, version)
+                    update_res.errors += res.errors
+                    update_res.updated.update(res.updated)
+                else:
+                    update_res.errors += check_tool_version(file, regexes, version).errors
+        else:
+            update_res.errors.append(f"No match with regex `{filename}`")
```

Comment:
A couple of suggestions: 
- Use "error first" pattern (treat "error" cases first to remove indentation levels) 
- Rename `filename` in `filepath_regex`
- Use `map` in the inner loop to get a list of `Path` objects

```suggestion
        matched_files = glob.glob(str(filepath_regex), recursive=True)
        if not matched_files:
            update_res.errors.append(f"No match with regex `{filepath_regex}`")
            continue

        # Check or update matched files
        for filepath in map(Path, matched_files):
            if update:
                res = update_tool_version(filepath, regexes, version)
                update_res.errors += res.errors
                update_res.updated.update(res.updated)
            else:
                update_res.errors += check_tool_version(filepath, regexes, version).errors
```

Reply:
See #7299